### PR TITLE
feat(partition): clip text in partition chart fill label

### PR DESCRIPTION
**Pull request by dgieselaar**
## Summary

Optionally clip text for fill label in partition chart, and fix an alignment issue.

Before:

image

After:
image

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [x] This was checked for cross-browser compatibility
- [x] Proper documentation or storybook story was added for features that require explanation or tutorials

### CODE REVIEW

File: src/chart_types/partition_chart/layout/viewmodel/fill_text_layout.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ function getVerticalAlignment(
     case VerticalAlignments.bottom:
       return -(container.y1 - linePitch * (totalRowCount - 1 - rowIndex) - paddingBottom - fontSize * overhang);
     default:
-      return -((container.y0 + container.y1) / 2 + (linePitch * (rowIndex - totalRowCount)) / 2);
+      return -((container.y0 + container.y1) / 2 + (linePitch * (rowIndex + 1 - totalRowCount)) / 2);
```
Comment (markov00) [inline src/chart_types/partition_chart/layout/viewmodel/fill_text_layout.ts:114]: :D

File: src/chart_types/partition_chart/renderer/canvas/canvas_renderers.ts [hunk 2]
```diff
@@ -58,6 +66,7 @@ function renderTextRow(
         ctx.font = cssFontShorthand(box, fontSize);
         ctx.fillText(box.text, box.width / 2 + box.wordBeginning, 0);
       });
+      ctx.closePath();
```
Comment (markov00) [inline src/chart_types/partition_chart/renderer/canvas/canvas_renderers.ts:69]: I don't think we need to call `closePath` here, and probably we don't even need the `ctx.beginPath();` at line 60 as `fillText` doesn't  need to instantiate a new path

File: src/chart_types/partition_chart/layout/config.ts [hunk 3]
```diff
@@ -212,6 +212,48 @@ export const configMetadata: Record<string, ConfigItem> = {
...
+            dflt: 2,
+            min: 0,
+            max: 20,
+            type: 'number',
+            reconfigurable: true,
+            documentation: 'Right padding for fill text',
+          },
+        },
+      },
+      clip: {
```
Comment (markov00) [inline src/chart_types/partition_chart/layout/config.ts:252]: we rename it `clipText` within the code as it clips the text, not the geometry. I know that is part of the `fillLabel` config, but we also have other attributes as `textColor` `textContrast` etc

**Review by markov00 — APPROVED** [review]
LGTM, I've left a few minor comments

File: src/chart_types/partition_chart/renderer/canvas/canvas_renderers.ts (hunk 2 above)
Comment (monfera) [reply src/chart_types/partition_chart/renderer/canvas/canvas_renderers.ts:69]: Indeed! The follow-up PR significantly restructures the Canvas render code, will explain reasons (motive: reduce context save/restore count)

File: src/chart_types/partition_chart/layout/config.ts (hunk 3 above)
Comment (monfera) [reply src/chart_types/partition_chart/layout/config.ts]: https://github.com/elastic/elastic-charts/pull/1033/commits/8853edff2457f22dad6dbe7b86ce13c4366ba46b

**Pull request merged**